peerpod-ctrl/Dockerfile

```diff
@@ -19,16 +19,17 @@ COPY main.go main.go
 COPY api/ api/
 COPY controllers/ controllers/
 
+RUN if [ "$TARGETARCH" = s390x ] ; then ln /usr/bin/gcc /usr/bin/s390x-linux-gnu-gcc; fi
```

Comment:
I haven't seen this as a requirement before for cross-compiling go programs

Reply:
We got error like this without the change https://github.com/confidential-containers/cloud-api-adaptor/actions/runs/6184725451/job/16788917938:
```
#18 180.2 cgo: C compiler "s390x-linux-gnu-gcc" not found: exec: "s390x-linux-gnu-gcc": executable file not found in $PATH
```
I guess the reason is GHA is doing cross-platform build while the golang image `quay.io/confidential-containers/golang-fedora:1.20.8-36` is s390x, thus, gcc is real `gcc` rather than emulator `s390x-linux-gnu-gcc` in this image.

Reply:
Success record is https://github.com/confidential-containers/cloud-api-adaptor/actions/runs/6185637431 with the change.